src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -292,5 +309,121 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            if (!TryReadFile(statusFilePath, reusableReader, out string fileContents))
+            {
+                // Between the time that we get an ID and the time that we try to read the associated stat
+                // file(s), the process could be gone.
+                result = default(ParsedStatus);
+                return false;
+            }
+
+            var results = default(ParsedStatus);
+            var statusFileContents = fileContents.AsSpan();
+            while (!statusFileContents.IsEmpty)
+            {
+                int startIndex = statusFileContents.IndexOf(':');
+                // Reached end of file
+                if (startIndex == -1)
+                {
+                    break;
+                }
+
+                ReadOnlySpan<char> title = statusFileContents.Slice(0, startIndex);
+                statusFileContents = statusFileContents.Slice(startIndex + 1);
+                int endIndex = statusFileContents.IndexOf('\n');
+                if (endIndex == -1)
+                {
+                    endIndex = statusFileContents.Length - 1;
```

Comment:
Shouldn't this be:
```suggestion
                    endIndex = statusFileContents.Length;
```
?

Reply:
Since its index, hence `length -1`.

Reply:
But you're using it like a length. For example, when you parse the value for Pid you do this:
```C#
value = statusFileContents.Slice(0, endIndex);
```
Let's say the string was "789\n"; the index returned for \n will be 3, and so you pass 3 as the count argument to slice.  Now let's say the string was "789".  You still need the count you pass in to be 3, which is Length, not Length - 1.

Reply:
>But you're using it like a length
Agreed. Wrong naming and corresponding usage. Refactored this a bit. Please check.